File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum = 0, maax = INT_MIN;
        for (int num : nums) {
            sum += num;
            maax = max(maax, sum);
            
            if (sum < 0)
                sum = 0;
        }
        return maax;
    }
};

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = 0, ans = Integer.MIN_VALUE;
        
        for (int n : nums) {
            max += n;
            ans = Math.max(ans, max);
            
            if (max < 0)
                max = 0;
        }
        return ans;
    }
}